www/js/controller.js

```diff
@@ -1072,6 +1072,29 @@ define([
 				}));
 
 			}
+		},
+		/**
+		* creates the object to be logged after each solution step.
+		* @params - obj -   object to be logged. Important parameter in the object is property of the node.
+		*                   without which the logged object wont be sent.
+		*           return obj - complete object which should be used by aspect.after to send it to logs.
+		*/
+		logSolutionStep: function(obj){
+			//Stub for logging the messages, updated in con-author and con-student
+			var property = obj.property;
+			// if node ID was not in the object then send node ID as the current ID.
+			// used for auto created nodes.
+			if(!obj.nodeID){
```

Comment:
Since we're going to return null if property is null, I think this if statement should get moved into the first clause of the next if statement, and then return obj should get moved to the end of that.  So the if(property) part either returns null, or does all the stuff and returns obj.